src/swap/components/SwapSettingsSlippageInput.tsx

```diff
@@ -0,0 +1,103 @@
+import { useState } from 'react';
+import { background, border, cn, color, pressable } from '../../styles/theme';
+import type { SwapSettingsSlippageInputReact } from '../types';
+import { useSwapContext } from './SwapProvider';
+
+export function SwapSettingsSlippageInput({
+  className,
+  defaultSlippage = 3,
+}: SwapSettingsSlippageInputReact) {
+  const { setLifeCycleStatus } = useSwapContext();
+  const [slippage, setSlippage] = useState(defaultSlippage);
+  const [isAutoMode, setIsAutoMode] = useState(true);
+
+  const updateSlippage = (newSlippage: number) => {
+    setSlippage(newSlippage);
+    setLifeCycleStatus({
+      statusName: 'slippageChange',
+      statusData: { maxSlippage: newSlippage },
+    });
+  };
+
+  const handleSlippageChange = (newSlippage: string) => {
+    const newSlippageNumber = Number.parseFloat(newSlippage);
+    if (!Number.isNaN(newSlippageNumber)) {
+      updateSlippage(newSlippageNumber);
+    }
+  };
+
+  const handleModeChange = (auto: boolean) => {
+    setIsAutoMode(auto);
+    if (auto) {
+      updateSlippage(defaultSlippage);
+    }
+  };
+
+  return (
+    <div className={className}>
+      <div
+        className={cn(
+          background.default,
+          border.defaultActive,
+          'flex items-center gap-2',
+          className,
+        )}
+      >
+        <div
+          className={cn(
+            background.default,
+            border.defaultActive,
+            'flex h-9 flex-1 rounded-xl border p-1',
+          )}
+        >
+          {['Auto', 'Custom'].map((mode) => (
```

Comment:
This list feels like a `const` that should be set at the top of the file.

Reply:
updated to `const slippageSettings = ['Auto', 'Custom'];`